lib/players/computer.rb

```diff
@@ -0,0 +1,124 @@
+class Computer < Player
+  attr_accessor :opp_token, :valid_moves
+
+  def move(board)
+    self.valid_moves = ["1", "2", "3", "4", "5", "6", "7", "8", "9"]
+    if self.token == "X"
+      self.opp_token = "O"
+    else
+      self.opp_token = "X"
+    end
+    if self.offensive(board)
+      self.offensive(board)
+    elsif self.defensive(board)
+      self.defensive(board)
+    else
+      self.valid_moves.sample
+    end
+  end
+
+  def offensive(board)
```

Comment:
lol i realized that a little ways into active record, i could of made a method for this :P that basically looked like...

def method_name(num1,num2,num3)
  if board.cells[num1] == self.token && board.cells[num2] == self.token && board.cells[num3] != self.opp_token
     self.valid_moves[num3]
   elsif board.cells[num1] == self.token && board.cells[num3] == self.token && board.cells[num2] != self.opp_token 
     self.valid_moves[num2]
   elsif board.cells[num2] == self.token && board.cells[num3] == self.token && board.cells[num1] != self.opp_token
    self.valid_moves[num1]
end

then called this method into the offensive method

def offensive_method
  method_name(0,1,2)
  method_name(3,4,5)
  method_name(6,7,8)
end

same logic with the defensive :)